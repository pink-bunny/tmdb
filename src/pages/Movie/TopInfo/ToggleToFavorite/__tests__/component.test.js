import React from 'react';
import { shallow } from 'enzyme';

import ToggleToFavoriteComponent from '../component';

describe('ToggleToFavorite component', () => {
  const requiredProps = {
    favorite: false,
    changeFavoriteStatus: jest.fn()
  };
  const wrapper = shallow(<ToggleToFavoriteComponent {...requiredProps} />); /* eslint-disable-line react/jsx-props-no-spreading, max-len */

  it('matches snapshot with required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot when favorite is turned on', () => {
    wrapper.setProps({ favorite: true });

    expect(wrapper).toMatchSnapshot();
  });
});
