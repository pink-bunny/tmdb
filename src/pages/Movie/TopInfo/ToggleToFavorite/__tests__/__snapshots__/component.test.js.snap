// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleToFavorite component matches snapshot when favorite is turned on 1`] = `
<Icon
  onClick={[MockFunction]}
  theme="filled"
  type="heart"
/>
`;

exports[`ToggleToFavorite component matches snapshot with required props 1`] = `
<Icon
  onClick={[MockFunction]}
  type="heart"
/>
`;
